crates/walrus-service/src/client/cli/args.rs

```diff
@@ -1173,50 +1164,56 @@ pub struct QuiltPatchQuery {
 impl QuiltPatchQuery {
     /// Returns a QuiltPatchSelector from the command line arguments.
     pub fn to_selector(&self) -> Result<QuiltPatchSelector> {
-        if !self.identifiers.is_empty() {
-            if !self.tag.is_empty() || !self.quilt_patch_ids.is_empty() || self.quilt_id.is_none() {
-                return Err(Self::invalid_query_error());
-            }
-            Ok(QuiltPatchSelector::ByIdentifier(QuiltPatchByIdentifier {
-                quilt_id: self.quilt_id.expect("quilt_id should be present"),
-                identifiers: self.identifiers.clone(),
-            }))
-        } else if !self.tag.is_empty() {
-            if self.tag.len() != 2 {
-                return Err(anyhow!("Only one tag is supported for now."));
-            }
-            if self.quilt_id.is_none() || self.quilt_patch_ids.is_empty() {
-                return Err(Self::invalid_query_error());
+        match (
+            !self.identifiers.is_empty(),
+            !self.tag.is_empty(),
+            !self.quilt_patch_ids.is_empty(),
+            self.quilt_id.is_some(),
```

Comment:
Nit: If we match on `self.quilt_id`, we can match `Some(quilt_id)` instead of having to extract it with `.expect("quilt_id should be present")`.